### PR TITLE
Add Lolin S2 Mini board

### LINKED ISSUE #1626: Support for Wemos S2-Mini (ESP32-S2FN4R2)
**Describe the problem you have/What new integration you would like**

I wish to use the Wemos S2-mini with esphome as it is a very small module (similar to the Wemos D1 mini with more pins)

**Please describe your use case for this integration and alternatives you've tried:**

I have tried all possible variants currently available for the ESP32-S2 and I keep getting the same error "Configuration does not match the platform of the connected device. Expected an ESP32 device."

**Additional context**

I assume this is because there isnt a platformio board type in existence for this S2-Mini yet? I have bumped a feature request (issue) in platformio github requesting S2-Mini support also. 

A link to the Wemos S2 Mini can be found here [https://www.wemos.cc/en/latest/s2/s2_mini.html](https://www.wemos.cc/en/latest/s2/s2_mini.html)

### PR DESCRIPTION
**Pull request by jhamhader**
Add ESP32-S2 Lolin/Wemos S2 Mini board.

Pins are taken from the official:
https://www.wemos.cc/en/latest/_static/files/sch_s2_mini_v1.0.0.pdf

# What does this implement/fix?

Add support ESP32-S2 Lolin/Wemos S2 Mini board.
Note, this depends on https://github.com/platformio/platform-espressif32/pull/736

Tested locally with a real Lolin/Wemos S2 Mini.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes [<link to issue>](https://github.com/esphome/feature-requests/issues/1626)

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** (TODO) esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
esp32:
  board: lolin_s2_mini
  type: esp-idf
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by jesserockz — APPROVED** [review]
Thanks

File: esphome/components/esp32/boards.py [hunk 1]
```diff
@@ -523,6 +523,24 @@
     "lolin32_lite": {"LED": 22},
     "lolin_d32": {"LED": 5, "_VBAT": 35},
     "lolin_d32_pro": {"LED": 5, "_VBAT": 35},
+    "lolin_s2_mini": {
```
Comment (Noltari) [inline esphome/components/esp32/boards.py:553]: According to the schematic LED is on GPIO15, so it can also be added here...
```
"LED": 15
```

Comment (joegross) [reply esphome/components/esp32/boards.py:553]: @jhamhader Looks like the merge may be blocked on adding the above addition?

**Comment by drapergeek** [comment]
Is there any momvement on this PR? I don't see any progress or issues.

**Comment by derekyle** [comment]
Agreed, it would be nice to have this merged in. It looks like the pull request it was waiting on in the platformio repo has been merged in.

**Comment by mkrsn** [comment]
@jhamhader thx for you Pull Request. I'm also waiting for this support. :dancers: 

It looks like the PR is blocked because there is the LED declaration missing in esphome/components/esp32/boards.py
Please see "Files Changed" comment from Noltari - line 526.

Would you please be so kind and add that? Thank you :)

**Comment by ssieb** [comment]
This is probably still waiting for an upstream release and for esphome to move to that new version.

**Comment by mkrsn** [comment]
@ssieb do you mean his PR https://github.com/platformio/platform-espressif32/pull/736 ?

This was already merged with v4.3.0 (latest: 5.1.1). Not sure what esphome currently use. But even if <v4.3.0, then it would be possible to go on without the recommended platform version and test at least.

**Comment by LarsStegman** [comment]
@jesserockz Any chance this can get merged? I want to use these boards for bluetooth proxying HomeKit door contact sensors.

**Comment by mkrsn** [comment]
@Noltari can you please confirm that the PR is only blocked because of the missing LED declaration?

In this case: The PR Owner seems to be unreachable or not willing to add the missing declaration. Is it ok if i redo the PR so that we get the new board on track?

**Comment by Noltari** [comment]
> @Noltari can you please confirm that the PR is only blocked because of the missing LED declaration?
> 
> In this case: The PR Owner seems to be unreachable or not willing to add the missing declaration. Is it ok if i redo the PR so that we get the new board on track?

@mkrsn there has been a misunderstanding because I'm not blocking anything at all.
I just made a suggestion about adding the missing LED, but this can be merged and the LED can be added afterwards if @jhamhader can't be reached.

**Comment by jhamhader** [comment]
I'll update the PR soon and we can merge :)

File: esphome/components/esp32/boards.py (hunk 1 above)
Comment (jhamhader) [reply esphome/components/esp32/boards.py:553]: Done

**Comment by jhamhader** [comment]
Rebased, fixed LED pin.

Not sure the default version includes https://github.com/platformio/platform-espressif32/pull/736

**Comment by jhamhader** [comment]
@jesserockz can you please take a look?

**Comment by Duckle29** [comment]
I believe this is blocked because esphome uses V3.5 of platform-espressif32. s2 mini support wasn't added until 4.3, so my guess is this is waiting for esphome to move to at least that.

Any official word on if this is the hangup?

https://github.com/esphome/esphome/pull/3807

**Review by jesserockz — APPROVED** [review]

**Pull request merged**